core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -319,41 +279,47 @@ wasi_addr_ip_to_bh_ip_addr_buffer(__wasi_addr_ip_t *addr,
     }
 }
 
+bh_clock_id_t
+convert_clock_to_bh_clock(__wasi_clockid_t in, bh_clock_id_t *out)
+{
+    switch (in) {
+
+        case __WASI_CLOCK_MONOTONIC:
+            return BH_CLOCK_ID_MONOTONIC;
+        case __WASI_CLOCK_PROCESS_CPUTIME_ID:
+            return BH_CLOCK_ID_PROCESS_CPUTIME_ID;
+        case __WASI_CLOCK_REALTIME:
+            return BH_CLOCK_ID_REALTIME;
+        case __WASI_CLOCK_THREAD_CPUTIME_ID:
+            return BH_CLOCK_ID_THREAD_CPUTIME_ID;
+        default:
+            return convert_errno(errno);
+    }
+}
+
 __wasi_errno_t
 wasmtime_ssp_clock_res_get(__wasi_clockid_t clock_id,
                            __wasi_timestamp_t *resolution)
 {
-#ifdef BH_PLATFORM_WINDOWS
-    return __WASI_ENOSYS;
-#else
-    clockid_t nclock_id;
-    if (!convert_clockid(clock_id, &nclock_id))
-        return __WASI_EINVAL;
-    struct timespec ts;
-    if (clock_getres(nclock_id, &ts) < 0)
+    bh_clock_id_t bh_clockdid;
+    if (convert_clock_to_bh_clock(!clock_id, &bh_clockdid))
```

Comment:
```suggestion
    if (!convert_clock_to_bh_clock(!clock_id, &bh_clockdid))
```

Reply:
Also, this should probably be 
```suggestion
    if (!convert_clock_to_bh_clock(!clock_id, &bh_clockdid))
```
if we update `convert_clock_to_bh_clock` to return `bool`